libres/tests/enkf/test_obs_data.cpp

```diff
@@ -0,0 +1,111 @@
+#include <string>
+
+#include "catch2/catch.hpp"
+
+#include <ert/enkf/obs_data.hpp>
+
+// std::normal_distribution has a different implementation depending on platform,
+// which makes this differentiation necessary.
+// See following issue for details:
+// https://github.com/equinor/ert/issues/2351#issuecomment-972880776
+#if defined(__linux__)
+std::string PLATFORM_NAME = "linux";
+#elif defined(__APPLE__)
+std::string PLATFORM_NAME = "macos";
+#endif
+
+SCENARIO(
+    "E-matrix is initialized using a normal distribution and scaled with data",
+    "[obs_data_allocE]") {
+    GIVEN("A obs_data with one obs_block and an instance of rng") {
+        double global_std_scaling = 1.0;
+        obs_data_type *obs_data = obs_data_alloc(global_std_scaling);
+
+        const char *obs_key = "obs_block_0";
+        const int obs_size = 3;
+        // error_covar does not effect this test, but is needed to initialize obs_block.
+        matrix_type *error_covar = matrix_alloc(obs_size, obs_size);
+        bool error_covar_owner = false;
+        obs_block_type *obs_block = obs_data_add_block(
+            obs_data, obs_key, obs_size, error_covar, error_covar_owner);
+        obs_block_iset(obs_block, 0, 5.0, 0.3);
+        obs_block_iset(obs_block, 2, 15.0, 0.5);
+
+        REQUIRE(obs_data_get_active_size(obs_data) == 2);
+        REQUIRE(obs_data_get_num_blocks(obs_data) == 1);
+
+        auto rng = rng_alloc(MZRAN, INIT_DEFAULT);
+
+        WHEN("E is allocated") {
+            matrix_type *E = obs_data_allocE(obs_data, rng, 2);
+
```

Comment:
Why not the == operator?

Reply:
Trying to target the line

`if (PLATFORM_NAME.compare("macos") == 0) {`

with this comment...